textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -0,0 +1,166 @@
+     *           matches. Returns only one tuple per document. If there are
+     *           multiple matches, all spans are included in a list. Java Regex
+     *           is used to match word boundaries. Ex : If 'String' name is "lin merry" and
+     *           the description 'Text' is
+     *           "Lin is like Angelina and is a merry person" and the Query is "Lin merry",
+     *           matches should include "lin merry","Lin","merry" but not Angelina.
+     */
+    @Override
+    public ITuple getNextTuple() throws DataFlowException {
+
+        List<IField> fieldList;
+        boolean foundFlag = false;
+        int positionIndex = 0; // Next position in the field to be checked.
+        int spanStartPosition; // Starting position of the matched query
+
+        try {
+            ITuple sourceTuple = sourceOperator.getNextTuple();
+            if(sourceTuple == null){
+                return null;
+            }
+            fieldList = sourceTuple.getFields();
+            spanList.clear();
+            if(!spanSchemaDefined){
+                spanSchemaDefined = true;
+                Schema schema = sourceTuple.getSchema();
+                spanSchema = Utils.createSpanSchema(schema);
+            }
+
+            for(int attributeIndex = 0; attributeIndex < attributeList.size(); attributeIndex++){
+                IField field = sourceTuple.getField(attributeList.get(attributeIndex).getFieldName());
+                String fieldValue = (String) (field).getValue();
+                String fieldName;
+                if(field instanceof StringField){
+                    //Keyword should match fieldValue entirely
+                    if(fieldValue.equals(query.toLowerCase())){
+                        spanStartPosition = 0;
+                        positionIndex = query.length();
+                        fieldName = attributeList.get(attributeIndex).getFieldName();
+                        addSpanToSpanList(fieldName, spanStartPosition, positionIndex, query, fieldValue);
+                        foundFlag = true;
+                    }
+                }
+                else if(field instanceof TextField) {
+                    //Each element of Array of keywords is matched in tokenized TextField Value
+                    for(int iter = 0; iter < queryTokens.size(); iter++) {
+                        positionIndex = 0;
+                        String queryToken = queryTokens.get(iter);
+                        //Ex: For keyword lin it obtains pattern like /blin/b which matches keywords at boundary
+                        Pattern tokenPattern = tokenPatternList.get(iter);
+                        Matcher matcher = tokenPattern.matcher(fieldValue.toLowerCase());
+                        while (matcher.find(positionIndex) != false) {
+                            spanStartPosition = matcher.start();
+                            positionIndex = spanStartPosition + queryTokens.get(iter).length();
```

Comment:
Sorry I was not clear.  In line 118 we have `String queryToken = queryTokens.get(iter);`.  So instead of using `queryTokens.get(iter)` in line 124, we can just use `queryToken`, right?